### ISSUE TITLE
--run exit code is wrong

### ISSUE THREAD

**Issue by LukeShu** [telepresenceio/telepresence#886]
What happened:

```
$ kubernaut claims create --name=lukeshu --cluster-group main
…
$ export KUBECONFIG=/home/lukeshu/.kube/lukeshu.yaml
$ telepresence --run sh -c 'exit 7'
T: Invoking sudo. Please enter your sudo password.
[sudo] password for lukeshu: 
T: Starting proxy with method 'vpn-tcp', which has the following limitations: All processes are affected, only one telepresence can run per machine, and you can't use other VPNs. You may need to add cloud hosts and headless services with --also-proxy. For a full list of 
T: method limitations see https://telepresence.io/reference/methods.html
T: Volumes are rooted at $TELEPRESENCE_ROOT. See https://telepresence.io/howto/volumes.html for details.

T: No traffic is being forwarded from the remote Deployment to your local machine. You can use the --expose option to specify which ports you want to forward.

T: Exit cleanup in progress
$ echo $?
0
```

What I expected t falseo happen: The exit status of the run command to be propagated out, or at least printed:

```
…
T: Main process (sh -c exit 7) exited with code 7.
$ echo $?
7
```

**Comment by LukeShu** [comment]
Note that reading `runner.py`, I'm not entirely sure why it doesn't print `T: Main process (sh -c exit 7) exited with code 7.`--it looks to me like it should.

**Comment by ark3** [comment]
`self.write(...)` vs `self.show(...)`.

I don't agree that Telepresence should appear to fail if the user process exits with a non-zero status. However, I think it makes sense to show a non-zero status to the user (and not just mention it in the log file).

**Comment by ark3** [comment]
Okay, maybe Telepresence does subvert expectations by returning success. Telepresence behaves more like `ssh` and less like `sudo` or `docker run`.

**Comment by LukeShu** [comment]
"Wrong" might be a little strong, but propagating the exit code is definitely what users expect, *especially* if we're encouraging them to use Telepresence in CI.

And, FWIW:
```
$ ssh lukeshu.com 'sh -c "exit 7"'
$ echo $?
7
```

**Comment by plombardi89** [comment]
Seems like a good use for a flag `--propogate-exitcode`... Or an inverse of that depending on the preferred default.

**Comment by LukeShu** [comment]
If you want the exit status hidden, the user can say `--run sh -c "YOUR_COMMAND || true"`

**Comment by plombardi89** [comment]
While they could do that it's not necessarily obvious or user friendly. While I'm sure many users use shell, I'm not convinced many users are nearly as familiar with shell idioms and usage patterns.

**Comment by LukeShu** [comment]
I figure the intersection of "users who care about whether an exit code came from telepresence itself or the child process" and "users who are not familiar with the `|| true` trick" is pretty small.

**Comment by ark3** [comment]
Telepresence's current behavior is to exit with status 1 on a detected error, status 3 if a session is successfully started and then fails ("lost connection"), and status 0 otherwise. I don't know how important it is to preserve this behavior. What is the likelihood that anyone would depend on it, or could depend on it in any useful way?

**Comment by LukeShu** [comment]
I don't think that the current behavior is in the docs, so I'd say that the likely hood that anyone is depending on it is pretty low.

We could also take the [`flock(1)`](https://manpage.me/index.cgi?apropos=0&q=flock&sektion=1&manpath=CentOS+7.1&arch=default&format=html) approach, and add `--FOO-exit-code N` flags to be able to adjust what statuses teleproxy itself emits. Eg. `--failed-to-start-exit-code N (default 1)` and `--lost-connection-exit N (default 3)`.

**Comment by ark3** [comment]
I'm reluctant to add more options to the Telepresence command line without very careful consideration. Telepresence is already too complicated to launch. See also #455. 

I am coming around to the idea of having Telepresence exit status change to be more like that of `ssh`, which return 255 on error but otherwise returns the status of the executed command. Thoughts?

Users, if this is important to you, please up-vote this issue.

**Comment by LukeShu** [comment]
POSIX specifies that exit codes ≥126 (well, 127, 127, and ≥129) are to indicate that there was an error running a program, and that programs should avoid using them for other purposes. 126 is "command exists but is not `chmod +x`", 127 is "command not found", 128+n is "killed by a signal n".  Because POSIX reserves the entire 129-255 range for "killed by signal" exit codes, but no system has 128 signals, so it is "safe" to use the higher end of that range for program-specific "I failed to create a child process" exit codes, as `ssh` does.
